docs/docs/javascript/overview/eslint.mdx

```diff
@@ -0,0 +1,554 @@
+    name="ts_files",
+    sources=["**/*.ts"],
+)
+```
+
+### Target-specific Configuration
+
+You can skip ESLint for specific targets using the `skip_eslint` field:
+
+```python title="src/js/BUILD"
+js_sources(
+    name="legacy",
+    sources=["legacy/*.js"],
+    skip_eslint=True,  # Skip linting for legacy code
+)
+
+typescript_sources(
+    name="modern",
+    sources=["modern/*.ts"],
+    # ESLint will run on these files
+)
+```
+
+## Configuration Files
+
+### Flat Config (ESLint 9.0+)
+
+```javascript title="eslint.config.js"
+export default {
+  languageOptions: {
+    ecmaVersion: 2022,
+    sourceType: "module",
+  },
+  rules: {
+    "no-unused-vars": "error",
+    "quotes": ["error", "double"],
+    "semi": ["error", "always"],
+  },
+};
+```
+
+### Legacy Config
+
+```json title=".eslintrc.json"
+{
+  "env": {
+    "browser": true,
+    "es2022": true
+  },
+  "extends": ["eslint:recommended"],
+  "rules": {
+    "no-unused-vars": "error",
+    "quotes": ["error", "double"],
+    "semi": ["error", "always"]
+  }
+}
+```
+
+### Package.json Config
+
+```json title="package.json"
+{
+  "name": "my-project",
+  "eslintConfig": {
+    "extends": ["eslint:recommended"],
+    "rules": {
+      "no-unused-vars": "error"
+    }
+  }
+}
+```
+
+## Supported File Types
+
+ESLint in Pants supports:
+
+- JavaScript (`.js`, `.mjs`, `.cjs`)
+- TypeScript (`.ts`) - requires `@typescript-eslint/parser` and `@typescript-eslint/eslint-plugin`
+
+## Integration with Other Tools
+
+### Prettier Integration
+
+ESLint and Prettier can work together. If you're using both tools, consider:
+
+1. Using `eslint-config-prettier` to disable ESLint formatting rules
+2. Running Prettier first, then ESLint
+3. Using ESLint plugins like `eslint-plugin-prettier`
+
+```toml title="pants.toml"
+[GLOBAL]
+backend_packages.add = [
+  "pants.backend.experimental.javascript.lint.prettier",
+  "pants.backend.experimental.javascript.lint.eslint",
+]
+```
+
+### TypeScript Support
+
+For TypeScript projects, you need to configure the TypeScript ESLint parser and plugin. In Pants, add them via `extra_dev_dependencies`:
+
+```toml title="pants.toml"
+[eslint]
+extra_dev_dependencies = [
+    "@typescript-eslint/parser@^6.0.0",
+    "@typescript-eslint/eslint-plugin@^6.0.0"
+]
+```
+
+Then configure your ESLint config to use them:
+
+```javascript title="eslint.config.js"
+import typescriptEslint from "@typescript-eslint/eslint-plugin";
+import parser from "@typescript-eslint/parser";
+
+export default {
+  files: ["**/*.ts"],
+  languageOptions: {
+    parser: parser,
+  },
+  plugins: {
+    "@typescript-eslint": typescriptEslint,
+  },
+  rules: {
+    "@typescript-eslint/no-unused-vars": "error",
+    "@typescript-eslint/explicit-function-return-type": "warn",
+  },
+};
+```
+
+**Note**: ESLint can lint TypeScript but has limited formatting capabilities for TypeScript-specific syntax. For comprehensive TypeScript formatting, consider using Prettier alongside ESLint.
```

Comment:
I wonder if it might make sense to have one docs section on linting and formatting, like with Python - https://www.pantsbuild.org/2.27/docs/python/overview/linters-and-formatters - but we can leave it for now.